hera_cal/tests/test_abscal.py

```diff
@@ -661,6 +708,49 @@ def test_global_phase_slope_logcal_3D(self):
             calibrate_in_place(data, gains)
         np.testing.assert_array_almost_equal(np.linalg.norm([data[bl] - model[bl] for bl in data]), 0, 5)
 
+    def test_complex_phase_abscal(self):
+        antpos = hex_array(3, split_core=False, outriggers=0)
+        reds = redcal.get_reds(antpos)
+        transformed_antpos = redcal.reds_to_antpos(reds)
+        abscal._put_transformed_array_on_integer_grid(transformed_antpos)
+
+        data_bls = model_bls = [group[0] for group in reds]
+        transformed_b_vecs = np.rint([transformed_antpos[jj] - transformed_antpos[ii] for (ii, jj, pol) in data_bls]).astype(int)
+
+        model, data = {}, {}
+        ntimes, nfreqs = 1, 2
+
+        # Test that the data is calibrated properly after being moved in phase
+        phase_deg = np.random.normal(0, 1, (ntimes, nfreqs, transformed_b_vecs.shape[-1]))
+        for bi, bls in enumerate(data_bls):
+            model[bls] = np.ones((ntimes, nfreqs))
+            data[bls] = model[bls] * np.exp(-1j * np.sum(transformed_b_vecs[bi][None, None, :] * phase_deg, axis=-1))
```

Comment:
In theory, you shouldn't have to use the transformed vectors when "uncalibrating" here.

Also, it might be worth testing this on an array that has extra DoF... perhaps two widely separated hexagons, with a phase difference between then in addition to EW and NS phase gradients.

Reply:
Good point, I'll add that test.

Reply:
I think that using the transformed vectors to uncalibrate is a fine unit test. And if we just add `split_core=True` then the test array will have 4 degeneracies.

Reply:
I changed the test array to have 4 degeneracies.